src/script/util/Entropy.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Wire
+ * Copyright (C) 2022 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+export interface EntropyFrame {
+  x: number;
+  y: number;
+  t: number;
+}
+
+// calculate shannon entropy over a set of uint8 values
+export function shannonEntropy<T>(entropyData: Uint8Array): number {
+  const len = entropyData.length;
+  const frequencies = entropyData.reduce((freq: Map<number, number>, c: number) => {
+    freq.set(c, (freq.get(c) || 0) + 1);
+    return freq;
+  }, new Map<number, number>());
+  let sum = 0;
+  for (const f of frequencies.values()) {
+    sum -= (f / len) * Math.log2(f / len);
+  }
+  return sum;
+}
+
+// calculate the delta of every n-th element
+export function calculateDeltaValues(data: Uint8Array, n: number): Uint8Array {
+  const prev = Array<number | null>(3);
+  const result = new Array<number>();
+
+  data.forEach((value, index) => {
+    const i = index % n;
+    const prevValue = prev[i];
+    if (prevValue != null) {
+      result.push(Math.abs(value - prevValue));
+    }
+    prev[i] = value;
+  });
+  return new Uint8Array(result);
+}
+
+export class EntropyData {
+  readonly frames: EntropyFrame[];
```

Comment:
are we assuming that this will always be our entropy type? perhaps this may also have a generic option?